client/asset/eth/eth.go

```diff
@@ -2371,6 +2462,119 @@ func (eth *ETHWallet) checkForNewBlocks(reportErr func(error)) {
 			w.checkFindRedemptions()
 		}
 	}()
+	go func() {
+		for _, w := range eth.walletList() {
+			w.checkConfirmRedemptions()
+		}
+	}()
+}
+
+func (w *assetWallet) swapIsRedeemed(secretHash [32]byte, contractVersion uint32) (bool, error) {
+	swap, err := w.swap(w.ctx, secretHash, contractVersion)
+	if err != nil {
+		return false, err
+	}
+	return swap.State == dexeth.SSRedeemed, nil
+}
+
+// confirmRedemption checks one of the queued redemption requests.
+func (w *assetWallet) confirmRedemption(txHash common.Hash, req *confirmRedemptionRequest, currentTip int64) {
+	const (
+		txConfsNeededToConfirm               = 10
+		blocksToWaitBeforeCoinNotFound       = 5
+		blocksToWaitBeforeCheckingIfReplaced = 10
+	)
+
+	var blocksSinceRequestStart int64
+	if currentTip > req.startHeight {
+		blocksSinceRequestStart = currentTip - req.startHeight
+	}
+
+	tx, txBlock, err := w.node.getTransaction(w.ctx, txHash)
```

Comment:
If after 10 blocks the transaction is still not found, and the transaction was not mined meaning that the state of the swap is still `Filled`, then the transaction would be resubmitted, probably with the same nonce.

Reply:
It may be already mined, but the light node cannot find it. At least this is what I have observed.

Reply:
Ran today on testnet. This is about how it usually looks:
<details closed>
<summary>goerli TestContract</summary>

```
=== RUN   TestContract
=== RUN   TestContract/testAddressesHaveFunds
=== RUN   TestContract/testSwap
(*eth.SwapState)(0xc038442d80)({
 BlockHeight: (uint64) 0,
 LockTime: (time.Time) 1970-01-01 09:00:00 +0900 JST,
 Secret: ([32]uint8) (len=32 cap=32) {
  00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
 },
 Initiator: (common.Address) (len=20 cap=20) 0x0000000000000000000000000000000000000000,
 Participant: (common.Address) (len=20 cap=20) 0x0000000000000000000000000000000000000000,
 Value: (uint64) 0,
 State: (eth.SwapStep) none
})
=== RUN   TestContract/testInitiate
INFO [06-27|19:00:32.438] Submitted transaction                    hash=0xef9cb854127bac918f26bb97cfd2ae843022eaedff1c7ed0f6a41429c3c9284b from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=286 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=2,000,000,000
INFO [06-27|19:00:36.216] Imported new block headers               count=1    elapsed="185.007µs" number=7,127,303 hash=021729..7798c5
INFO [06-27|19:00:36.610] Imported new block headers               count=1    elapsed="148.308µs" number=7,127,303 hash=021729..7798c5
(*types.Receipt)(0xc038fc6700)({
 Type: (uint8) 2,
 PostState: ([]uint8) <nil>,
 Status: (uint64) 1,
 CumulativeGasUsed: (uint64) 424364,
 Bloom: (types.Bloom) (len=256 cap=256) {
  00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000040  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000070  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000080  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
 },
 Logs: ([]*types.Log) {
 },
 TxHash: (common.Hash) (len=32 cap=32) 0xef9cb854127bac918f26bb97cfd2ae843022eaedff1c7ed0f6a41429c3c9284b,
 ContractAddress: (common.Address) (len=20 cap=20) 0x0000000000000000000000000000000000000000,
 GasUsed: (uint64) 134458,
 BlockHash: (common.Hash) (len=32 cap=32) 0x0217290402c1fb062094fad259025ae74cc48db81e38febd6bb3d2c3e47798c5,
 BlockNumber: (*big.Int)(0xc038ff0420)(7127303),
 TransactionIndex: (uint) 2
})
Gas used for 1 initiations, success = true: 134458 (expected max 135000) 
INFO [06-27|19:00:47.448] Submitted transaction                    hash=0x9ae126fc9d9577c4f3a2e3ca496dc840f94c62de94ce3961ebe572258256d2f8 from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=287 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=1,000,000,000
INFO [06-27|19:00:53.722] Imported new block headers               count=1    elapsed="158.167µs" number=7,127,304 hash=2d87c3..e7d991
INFO [06-27|19:00:54.370] Imported new block headers               count=1    elapsed="190.797µs" number=7,127,304 hash=0d8113..5ec6a5
    nodeclient_harness_test.go:1072: 1 swap with existing hash: failed retrieving initiate receipt: transaction 0x9ae126fc9d9577c4f3a2e3ca496dc840f94c62de94ce3961ebe572258256d2f8 not found
=== RUN   TestContract/testRedeem
INFO [06-27|19:01:05.007] Submitted transaction                    hash=0x99bc5e04f15c01ba57020897a9e5f9f64b70d7fab69c8d83393df895c238448c from=0x1D4F2ee206474B136Af4868B887C7b166693c194 nonce=90  recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=1,000,000,000
INFO [06-27|19:01:06.659] Imported new block headers               count=2    elapsed="841.337µs" number=7,127,305 hash=9d2071..98fcc4 ignored=1
INFO [06-27|19:01:07.038] Imported new block headers               count=1    elapsed="144.451µs" number=7,127,305 hash=9d2071..98fcc4
INFO [06-27|19:01:21.181] Imported new block headers               count=1    elapsed="131.286µs" number=7,127,306 hash=129b4e..bd1cc9
INFO [06-27|19:01:22.096] Imported new block headers               count=1    elapsed="110.487µs" number=7,127,306 hash=129b4e..bd1cc9
INFO [06-27|19:01:37.124] Imported new block headers               count=1    elapsed="158.718µs" number=7,127,307 hash=b9947c..8b340f
INFO [06-27|19:01:37.325] Imported new block headers               count=1    elapsed="158.207µs" number=7,127,307 hash=b9947c..8b340f
INFO [06-27|19:01:51.469] Imported new block headers               count=1    elapsed="162.735µs" number=7,127,308 hash=c3bc7f..c6631c
INFO [06-27|19:01:51.765] Imported new block headers               count=1    elapsed="157.395µs" number=7,127,308 hash=c3bc7f..c6631c
INFO [06-27|19:02:06.684] Imported new block headers               count=1    elapsed="144.43µs"  number=7,127,309 hash=97ad96..d9b7de
INFO [06-27|19:02:06.945] Imported new block headers               count=1    elapsed="172.544µs" number=7,127,309 hash=97ad96..d9b7de
INFO [06-27|19:02:21.277] Imported new block headers               count=1    elapsed="164.569µs" number=7,127,310 hash=f39707..44babf
INFO [06-27|19:02:21.674] Imported new block headers               count=1    elapsed="135.794µs" number=7,127,310 hash=f39707..44babf
INFO [06-27|19:02:35.911] Imported new block headers               count=1    elapsed="176.471µs" number=7,127,311 hash=c0390a..d405b5
INFO [06-27|19:02:36.175] Imported new block headers               count=1    elapsed="141.355µs" number=7,127,311 hash=c0390a..d405b5
INFO [06-27|19:02:53.433] Imported new block headers               count=1    elapsed="150.402µs" number=7,127,312 hash=8e385d..d48892
INFO [06-27|19:02:53.958] Imported new block headers               count=1    elapsed="152.045µs" number=7,127,312 hash=8e385d..d48892
    nodeclient_harness_test.go:1399: ok before locktime: failed to get init receipt: transaction 0x99bc5e04f15c01ba57020897a9e5f9f64b70d7fab69c8d83393df895c238448c not found
=== RUN   TestContract/testRefund
INFO [06-27|19:03:05.307] Submitted transaction                    hash=0x9601e3d2e8d7f654c1697113898be0e8680fde89878b9add71bdecb4cc91286e from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=288 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=1,000,000,000
INFO [06-27|19:03:06.352] Imported new block headers               count=1    elapsed="141.495µs" number=7,127,313 hash=d6e3d0..1a02e0
INFO [06-27|19:03:06.724] Imported new block headers               count=1    elapsed="155.341µs" number=7,127,313 hash=d6e3d0..1a02e0
INFO [06-27|19:03:21.122] Imported new block headers               count=1    elapsed="145.322µs" number=7,127,314 hash=14568d..23eb58
INFO [06-27|19:03:21.629] Imported new block headers               count=1    elapsed="178.745µs" number=7,127,314 hash=14568d..23eb58
INFO [06-27|19:03:37.264] Imported new block headers               count=1    elapsed="164.689µs" number=7,127,315 hash=263a14..4a0975
INFO [06-27|19:03:37.528] Imported new block headers               count=1    elapsed="163.607µs" number=7,127,315 hash=263a14..4a0975
INFO [06-27|19:03:51.106] Imported new block headers               count=1    elapsed="165.06µs"  number=7,127,316 hash=602764..6b9ad2
INFO [06-27|19:03:51.424] Imported new block headers               count=1    elapsed="84.599µs"  number=7,127,316 hash=602764..6b9ad2
INFO [06-27|19:04:06.401] Imported new block headers               count=1    elapsed="145.763µs" number=7,127,317 hash=b9a597..889e40
INFO [06-27|19:04:07.544] Imported new block headers               count=1    elapsed="107.101µs" number=7,127,317 hash=b9a597..889e40
INFO [06-27|19:04:22.999] Imported new block headers               count=1    elapsed="183.113µs" number=7,127,318 hash=0f710d..7958e0
INFO [06-27|19:04:23.235] Imported new block headers               count=1    elapsed="148.178µs" number=7,127,318 hash=bf7529..daae83
INFO [06-27|19:04:36.292] Imported new block headers               count=1    elapsed="184.506µs" number=7,127,319 hash=f4baf0..325b18
INFO [06-27|19:04:37.433] Imported new block headers               count=2    elapsed="834.034µs" number=7,127,319 hash=f4baf0..325b18 ignored=1
INFO [06-27|19:04:52.872] Imported new block headers               count=1    elapsed="182.251µs" number=7,127,320 hash=b6142b..29a7dd
INFO [06-27|19:04:53.182] Imported new block headers               count=1    elapsed="113.733µs" number=7,127,320 hash=b6142b..29a7dd
INFO [06-27|19:05:05.316] Submitted transaction                    hash=0xa0721518402b8759fc038260ade71775458bd4b8e03371d9699a3877c20c6a0a from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=289 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=0
(*types.Transaction)(0xc00332ade0)({
 inner: (*types.DynamicFeeTx)(0xc030c38e80)({
  ChainID: (*big.Int)(0xc000395580)(5),
  Nonce: (uint64) 289,
  GasTipCap: (*big.Int)(0xc03f8a04c0)(2000000000),
  GasFeeCap: (*big.Int)(0xc03f8a04e0)(2000000000),
  Gas: (uint64) 43000,
  To: (*common.Address)(0xc05cc7ede0)((len=20 cap=20) 0xa483b6166dA8Da6748B29Af35f96C4F9388c456C),
  Value: (*big.Int)(0xc03f8a0480)(0),
  Data: ([]uint8) (len=36 cap=36) {
   00000000  72 49 fb b6 47 3d 50 6b  d7 4e ee 86 23 d8 c1 58  |rI..G=Pk.N..#..X|
   00000010  c7 db 69 b9 a0 ca 23 d9  16 94 7c 38 4b e5 9f 46  |..i...#...|8K..F|
   00000020  e7 d6 f0 84                                       |....|
  },
  AccessList: (types.AccessList) {
  },
  V: (*big.Int)(0xc03f8a0460)(1),
  R: (*big.Int)(0xc03f8a0400)(65034802707364872847383889061440937742239229432279143060947646781196239171176),
  S: (*big.Int)(0xc03f8a0420)(56825073346672081823344599504605177319440851474461699529971336906353830120699)
 }),
 time: (time.Time) 2022-06-27 19:05:05.315588696 +0900 JST m=+392.193411981,
 hash: (atomic.Value) {
  v: (interface {}) <nil>
 },
 size: (atomic.Value) {
  v: (interface {}) <nil>
 },
 from: (atomic.Value) {
  v: (interface {}) <nil>
 }
})
INFO [06-27|19:05:06.651] Imported new block headers               count=1    elapsed="174.136µs" number=7,127,321 hash=bf3aac..f924a2
INFO [06-27|19:05:06.858] Imported new block headers               count=1    elapsed="135.454µs" number=7,127,321 hash=bf3aac..f924a2
(*types.Receipt)(0xc00294cfc0)({
 Type: (uint8) 2,
 PostState: ([]uint8) <nil>,
 Status: (uint64) 1,
 CumulativeGasUsed: (uint64) 1292628,
 Bloom: (types.Bloom) (len=256 cap=256) {
  00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000040  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000070  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000080  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
 },
 Logs: ([]*types.Log) {
 },
 TxHash: (common.Hash) (len=32 cap=32) 0xa0721518402b8759fc038260ade71775458bd4b8e03371d9699a3877c20c6a0a,
 ContractAddress: (common.Address) (len=20 cap=20) 0x0000000000000000000000000000000000000000,
 GasUsed: (uint64) 40531,
 BlockHash: (common.Hash) (len=32 cap=32) 0xbf3aac8ab70f6a4c53006280305de1bbc4cd8847d42e527ce9f3d99407f924a2,
 BlockNumber: (*big.Int)(0xc01c0cc8a0)(7127321),
 TransactionIndex: (uint) 9
})
Gas used for refund, success = true: 40531 
INFO [06-27|19:05:20.331] Submitted transaction                    hash=0x1bd6b4d3269ecba775d64fa99c38d5a71a80abe3ec61ec9c81c4d94b1078d670 from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=290 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=1,000,000,000
INFO [06-27|19:05:22.155] Imported new block headers               count=1    elapsed="234.62µs"  number=7,127,322 hash=82d735..ac522a
INFO [06-27|19:05:22.505] Imported new block headers               count=1    elapsed="149.32µs"  number=7,127,322 hash=82d735..ac522a
INFO [06-27|19:05:37.654] Imported new block headers               count=1    elapsed="136.315µs" number=7,127,323 hash=217217..837fef
INFO [06-27|19:05:38.213] Imported new block headers               count=1    elapsed="176.791µs" number=7,127,323 hash=217217..837fef
INFO [06-27|19:05:52.627] Imported new block headers               count=1    elapsed="160.11µs"  number=7,127,324 hash=b13082..b033ee
INFO [06-27|19:05:52.986] Imported new block headers               count=1    elapsed="159.369µs" number=7,127,324 hash=b13082..b033ee
INFO [06-27|19:06:06.674] Imported new block headers               count=1    elapsed="185.147µs" number=7,127,325 hash=e43012..84f9ca
INFO [06-27|19:06:06.984] Imported new block headers               count=1    elapsed="149.781µs" number=7,127,325 hash=e43012..84f9ca
INFO [06-27|19:06:21.090] Imported new block headers               count=1    elapsed="198.793µs" number=7,127,326 hash=c61470..ce5711
INFO [06-27|19:06:21.465] Imported new block headers               count=1    elapsed="148.699µs" number=7,127,326 hash=c61470..ce5711
INFO [06-27|19:06:36.151] Imported new block headers               count=1    elapsed="150.913µs" number=7,127,327 hash=a7d997..120e92
INFO [06-27|19:06:36.475] Imported new block headers               count=1    elapsed="160.741µs" number=7,127,327 hash=a7d997..120e92
INFO [06-27|19:06:52.414] Imported new block headers               count=1    elapsed="213.4µs"   number=7,127,328 hash=ba442a..94f2ae
INFO [06-27|19:06:52.625] Imported new block headers               count=1    elapsed="188.293µs" number=7,127,328 hash=ba442a..94f2ae
INFO [06-27|19:07:06.731] Imported new block headers               count=1    elapsed="183.574µs" number=7,127,329 hash=fbb964..70cef3
INFO [06-27|19:07:07.098] Imported new block headers               count=1    elapsed="138.018µs" number=7,127,329 hash=fbb964..70cef3
INFO [06-27|19:07:20.339] Submitted transaction                    hash=0x41ad721013a067f7ec82e2490a5cacbf1daefe7c483374660a0df4e7652238fd from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=291 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=0
(*types.Transaction)(0xc04b55fa40)({
 inner: (*types.DynamicFeeTx)(0xc03fccdc00)({
  ChainID: (*big.Int)(0xc000395580)(5),
  Nonce: (uint64) 291,
  GasTipCap: (*big.Int)(0xc062ae93a0)(2000000000),
  GasFeeCap: (*big.Int)(0xc062ae93c0)(2000000000),
  Gas: (uint64) 43000,
  To: (*common.Address)(0xc03a8eb650)((len=20 cap=20) 0xa483b6166dA8Da6748B29Af35f96C4F9388c456C),
  Value: (*big.Int)(0xc062ae9360)(0),
  Data: ([]uint8) (len=36 cap=36) {
   00000000  72 49 fb b6 87 15 cc 71  36 27 83 fb 51 88 b0 c0  |rI.....q6'..Q...|
   00000010  98 cc 4a e2 e1 ba 5f 45  24 8a e3 58 48 89 e1 3a  |..J..._E$..XH..:|
   00000020  a0 f8 0b 8b                                       |....|
  },
  AccessList: (types.AccessList) {
  },
  V: (*big.Int)(0xc062ae9340)(0),
  R: (*big.Int)(0xc062ae92e0)(92606112962078848713910089198371527465006627527781069599214150132306875766933),
  S: (*big.Int)(0xc062ae9300)(24452402343336706551782301940370962335355727162224432614049394667483696863360)
 }),
 time: (time.Time) 2022-06-27 19:07:20.339210624 +0900 JST m=+527.217033909,
 hash: (atomic.Value) {
  v: (interface {}) <nil>
 },
 size: (atomic.Value) {
  v: (interface {}) <nil>
 },
 from: (atomic.Value) {
  v: (interface {}) <nil>
 }
})
INFO [06-27|19:07:22.799] Imported new block headers               count=1    elapsed="171.551µs" number=7,127,330 hash=17d435..4fd04a
INFO [06-27|19:07:23.107] Imported new block headers               count=1    elapsed="140.904µs" number=7,127,330 hash=17d435..4fd04a
(*types.Receipt)(0xc037c18fc0)({
 Type: (uint8) 2,
 PostState: ([]uint8) <nil>,
 Status: (uint64) 0,
 CumulativeGasUsed: (uint64) 538078,
 Bloom: (types.Bloom) (len=256 cap=256) {
  00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000040  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000070  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000080  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  00000090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
  000000f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
 },
 Logs: ([]*types.Log) {
 },
 TxHash: (common.Hash) (len=32 cap=32) 0x41ad721013a067f7ec82e2490a5cacbf1daefe7c483374660a0df4e7652238fd,
 ContractAddress: (common.Address) (len=20 cap=20) 0x0000000000000000000000000000000000000000,
 GasUsed: (uint64) 26354,
 BlockHash: (common.Hash) (len=32 cap=32) 0x17d43519f6bfd28ee9fcaed184893a37e454f2dca20b3221a3228db5304fd04a,
 BlockNumber: (*big.Int)(0xc0408ffc20)(7127330),
 TransactionIndex: (uint) 4
})
Gas used for refund, success = false: 26354 
INFO [06-27|19:07:35.352] Submitted transaction                    hash=0x67e5533a52fb361fb672637b2baf92d6eb586d40d9698b6e91e4fde197cff4fe from=0xdd93b447f7eBCA361805eBe056259853F3912E04 nonce=292 recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=1,000,000,000
INFO [06-27|19:07:36.896] Imported new block headers               count=1    elapsed="226.825µs" number=7,127,331 hash=eb4084..3857fd
INFO [06-27|19:07:37.197] Imported new block headers               count=1    elapsed="168.576µs" number=7,127,331 hash=eb4084..3857fd
INFO [06-27|19:07:51.046] Imported new block headers               count=1    elapsed="191.76µs"  number=7,127,332 hash=e23d80..e93abe
INFO [06-27|19:07:51.425] Imported new block headers               count=1    elapsed="154.309µs" number=7,127,332 hash=e23d80..e93abe
INFO [06-27|19:08:05.640] Submitted transaction                    hash=0xd23cb1200728964cd765cc69289d1324f1f379080234fd84377aa27dc11f79e1 from=0x1D4F2ee206474B136Af4868B887C7b166693c194 nonce=91  recipient=0xa483b6166dA8Da6748B29Af35f96C4F9388c456C value=0
INFO [06-27|19:08:07.609] Imported new block headers               count=1    elapsed="195.517µs" number=7,127,333 hash=5392d9..bf0231
INFO [06-27|19:08:07.899] Imported new block headers               count=1    elapsed="136.215µs" number=7,127,333 hash=5392d9..bf0231
INFO [06-27|19:08:22.260] Imported new block headers               count=1    elapsed="113.172µs" number=7,127,334 hash=37cbc8..f671f0
INFO [06-27|19:08:22.642] Imported new block headers               count=1    elapsed="139.662µs" number=7,127,334 hash=37cbc8..f671f0
INFO [06-27|19:08:36.727] Imported new block headers               count=1    elapsed="166.723µs" number=7,127,335 hash=9bb69a..9db3f7
INFO [06-27|19:08:36.868] Imported new block headers               count=1    elapsed="160.12µs"  number=7,127,335 hash=3666b4..c8de37
INFO [06-27|19:08:51.728] Imported new block headers               count=1    elapsed="173.135µs" number=7,127,336 hash=1c541b..b74da3
INFO [06-27|19:08:52.972] Imported new block headers               count=2    elapsed="563.877µs" number=7,127,336 hash=1c541b..b74da3 ignored=1
INFO [06-27|19:09:07.250] Imported new block headers               count=1    elapsed="167.514µs" number=7,127,337 hash=68782e..63adfd
INFO [06-27|19:09:07.633] Imported new block headers               count=1    elapsed="155.892µs" number=7,127,337 hash=68782e..63adfd
INFO [06-27|19:09:22.585] Imported new block headers               count=1    elapsed="143.228µs" number=7,127,338 hash=241409..de8ec3
INFO [06-27|19:09:22.783] Imported new block headers               count=1    elapsed="138.189µs" number=7,127,338 hash=241409..de8ec3
INFO [06-27|19:09:36.985] Imported new block headers               count=1    elapsed="180.759µs" number=7,127,339 hash=49c9e5..290ee4
INFO [06-27|19:09:37.254] Imported new block headers               count=1    elapsed="134.212µs" number=7,127,339 hash=49c9e5..290ee4
INFO [06-27|19:09:50.978] Imported new block headers               count=1    elapsed="117.761µs" number=7,127,340 hash=93210c..0cab66
INFO [06-27|19:09:51.322] Imported new block headers               count=1    elapsed="138.129µs" number=7,127,340 hash=93210c..0cab66
    nodeclient_harness_test.go:1721: already redeemed: expected isRefundable=false, but got true
--- FAIL: TestContract (573.25s)
    --- PASS: TestContract/testAddressesHaveFunds (0.00s)
    --- PASS: TestContract/testSwap (0.01s)
    --- FAIL: TestContract/testInitiate (30.05s)
    --- FAIL: TestContract/testRedeem (122.84s)
    --- FAIL: TestContract/testRefund (420.35s)
```
</details>

Actually more often than not it seems the tx is not mined. I think I'll make a proper issue on this. Do you see anything obvious why these would fail so often?

In one place at least it failed to get a tx receipt for a tx that was mined: https://goerli.etherscan.io/tx/0x9ae126fc9d9577c4f3a2e3ca496dc840f94c62de94ce3961ebe572258256d2f8

But you know, I don't think I've noticed tx not getting mined on testnet when testing actual trades.

I'll also test this pr out on testnet for a bit and see if it throws tx cant be found for the redeems.

Reply:
The tests do randomly pass.

Reply:
I apologize, my assumptions that the light node can't find txn seems to be false. It turns out the simnet tests were failing on testnet because we were looking at the chain from the point of view of an orphaned (uncle???) block randomly. After allowing some time to find the tx when we get back on the right chain, the tests pass without problems.

Reply:
This doesn't really explain what I saw in the linked issue on simnet... but I guess if you just look for the tx again it did find it. So, no problem really.

Reply:
OK cool, I had a new plan that worked without using `getTrasnaction` but I'll keep it the way it is in that case.